### PR TITLE
allow users to set envGracePeriod to 0

### LINKED ISSUE #2686: Environment spec config -- preStop not working as expected
**Fission/Kubernetes version**

<pre>
$ fission version v1.17.0
$ kubectl version v1.25.1
</pre>

**Kubernetes platform (e.g. Google Kubernetes Engine)**
**
**Describe the bug**

Spec config preStop is not working as expected, I want a fresh new preStop config be worked, but new preStop command was actually merge  into an old exec command list.
**To Reproduce**

```yaml
# Environment spec config
spec:
  runtime:
    container:
      lifecycle:
        preStop:
          exec:
            command: ["/bin/sleep", "0"] # here
```

**Expected result**

Get deploy pod config information by k8s
```yaml
spec:
  containers:
    image: xxxx
    imagePullPolicy: IfNotPresent
    lifecycle:
      preStop:
        exec:
          command:
          - /bin/sleep
          - "0"
    name: xxx
```

**Actual result**
```yaml
spec:
  containers:
    image: xxxx
    imagePullPolicy: IfNotPresent
    lifecycle:
      preStop:
        exec:
          command:
          - /bin/sleep
          - "5"
          - /bin/sleep
          - "0"
    name: xxx
```
**Screenshots/Dump file**

**Additional context**

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
Signed-off-by: Nikhl Sharma <nikhilsharma230303@gmail.com>

## Description

This PR allows users to set the value of the `environment grace period` to any non-negative integer.
## Which issue(s) this PR fixes:

Fixes #2686

## Checklist:

- [x] I ran tests as well as code linting locally to verify my changes. 
- [x] I have done manual verification of my changes, changes working as expected.
- [ ] I have added new tests to cover my changes.
- [x] My changes follow contributing guidelines of Fission.
- [x] I have signed all of my commits.

### CODE REVIEW

File: pkg/executor/executortype/newdeploy/newdeploy.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (deploy *NewDeploy) getDeploymentSpec(ctx context.Context, fn *fv1.Function
 	}
 
 	gracePeriodSeconds := int64(6 * 60)
-	if env.Spec.TerminationGracePeriod > 0 {
+	if !(env.Spec.TerminationGracePeriod < 0) {
```
Comment (sanketsudake) [inline pkg/executor/executortype/newdeploy/newdeploy.go:139]: Should we write it as `env.Spec.TerminationGracePeriod >= 0` ? same for other condition

File: pkg/fission-cli/flag/flag.go [hunk 2]
```diff
@@ -181,7 +181,7 @@ var (
 	EnvBuildCmd               = Flag{Type: String, Name: flagkey.EnvBuildcommand, Usage: "Build command for environment builder to build source package"}
 	EnvKeepArchive            = Flag{Type: Bool, Name: flagkey.EnvKeeparchive, Usage: "Keep the archive instead of extracting it into a directory (mainly for the JVM environment because .jar is one kind of zip archive)"}
 	EnvExternalNetwork        = Flag{Type: Bool, Name: flagkey.EnvExternalNetwork, Usage: "Allow pod to access external network (only works when istio feature is enabled)"}
-	EnvTerminationGracePeriod = Flag{Type: Int64, Name: flagkey.EnvGracePeriod, Aliases: []string{"period"}, Usage: "Grace time (in seconds) for pod to perform connection draining before termination (default value will be used if 0 is given)", DefaultValue: 360}
+	EnvTerminationGracePeriod = Flag{Type: Int64, Name: flagkey.EnvGracePeriod, Aliases: []string{"period"}, Usage: "Grace time (in seconds) for pod to perform connection draining before termination (default value will be used if negative value is given)", DefaultValue: 360}
```
Comment (sanketsudake) [inline pkg/fission-cli/flag/flag.go:184]: `(default value will be used if negative value is given`
Lets remove this.  

we can write this as 
`Grace time (in seconds) for pod to perform connection draining before termination (only non-negative values considered)"`

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: pkg/executor/executortype/newdeploy/newdeploy.go (hunk 1 above)
Comment (NikhilSharmaWe) [reply pkg/executor/executortype/newdeploy/newdeploy.go:139]: Ok

File: pkg/fission-cli/flag/flag.go (hunk 2 above)
Comment (NikhilSharmaWe) [reply pkg/fission-cli/flag/flag.go:184]: Also change this for fngraceperiod, right?

**Review by sanketsudake — APPROVED** [review]

**Review by shubham-bansal96 — APPROVED** [review]

**Pull request merged**